replica/rpc/server.go

```diff
@@ -51,12 +54,47 @@ func (s *Server) ListenAndServe() error {
 			continue
 		}
 
-		logrus.Infof("New connection from: %v", conn.RemoteAddr())
+		if len(s.s.ClientAddr) == 0 {
+			logrus.Infof("New connection from: %v", conn.RemoteAddr())
+			s.s.ClientAddr = conn.RemoteAddr().String()
```

Comment:
let us set clientAddr during handlePing of rpc/server.go, and use that address for comparison here.